.github/workflows/ci.yml

```diff
@@ -440,3 +440,49 @@ jobs:
         run: |
           echo "Testing against ${RELAY_TEST_IMAGE}"
           make test-relay-integration
+
+  publish-to-dockerhub:
+    name: Publish Relay to DockerHub
+    needs: build
+    runs-on: ubuntu-20.04
+    if: ${{ (github.ref_name == 'master') }}
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+      - name: Pull the test image
+        id: image_pull
+        env:
+          IMAGE_URL: ghcr.io/getsentry/relay:${{ github.sha }}
+        shell: bash
+        run: |
+          echo "We poll for the Docker image that the GCB/GHA build produces until it succeeds or this job times out."
+          echo "Polling for $IMAGE_URL"
+          timeout 20m bash -c 'until docker pull "$IMAGE_URL" 2>/dev/null; do sleep 10; done'
```

Comment:
- Could you configure a job or step timeout rather than using the `timeout` command here? 
- The `echo` seems redundant since we're not gonna look at that; let's optimize for less verbose actions please.
- Could we inline `IMAGE_URL` instead of declaring it as an env, to make the action smaller?
- Remove the step `id` since it's not needed. I'd even go as far as remove the step name, since it's clear what this does.
- The shell is not needed. 

To put this together, we could write this step as:

```yaml
steps:
   - run: docker pull "ghcr.io/getsentry/relay:${{ github.sha }}"
```

I'm making these suggestions because our CI has become complex and hard to comprehend. We should take great care to write it in a concise way.